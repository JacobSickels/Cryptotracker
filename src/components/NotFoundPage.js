import React from 'react';
import { Link } from 'react-router-dom';
import Header from './Header';
import Footer from './Footer';

/*
    NotFoundPage is responsible for showing an error message when
    a page is navigated to that doesn't exist in AppRouter. In this case, 
    a dummy 404 message is shown and prompts the user to go back to the Dashboard.
*/

const NotFoundPage = () => (
    <div>
        <Header />
            <div className="page-container">
                <div className="container">
                    <h1>We're Sorry! This page doesn't exist </h1>
                    <p>Error Code: 404 - <Link to="/">Go Home</Link></p>
                </div>
            </div>
        <Footer />
    </div>
);

export default NotFoundPage;