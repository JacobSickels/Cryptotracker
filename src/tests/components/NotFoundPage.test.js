import React from 'react';
import { shallow } from 'enzyme';
import NotFoundPage from '../../components/NotFoundPage';

//This test makes sure the NotFoundPage component is rendered correctly
test('Should render NotFoundPage correctly', () => {
    const wrapper = shallow(<NotFoundPage />);
    expect(wrapper).toMatchSnapshot();
});