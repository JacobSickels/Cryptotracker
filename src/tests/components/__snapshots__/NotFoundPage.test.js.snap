// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render NotFoundPage correctly 1`] = `
<div>
  <Connect(Header) />
  <div
    className="page-container"
  >
    <div
      className="container"
    >
      <h1>
        We're Sorry! This page doesn't exist 
      </h1>
      <p>
        Error Code: 404 - 
        <Link
          replace={false}
          to="/"
        >
          Go Home
        </Link>
      </p>
    </div>
  </div>
  <Footer />
</div>
`;
